geode-common/src/main/java/org/apache/geode/internal/Retry.java

```diff
@@ -32,18 +31,21 @@
   interface Timer {
     long nanoTime();
 
-    void sleep(long sleepTime, TimeUnit sleepTimeUnit) throws InterruptedException;
+    void sleep(long sleepTimeInNano) throws InterruptedException;
   }
 
-  private static class SteadyTimer implements Timer {
+  static class SteadyTimer implements Timer {
     @Override
     public long nanoTime() {
       return System.nanoTime();
     }
 
     @Override
-    public void sleep(long sleepTime, TimeUnit sleepTimeUnit) throws InterruptedException {
-      Thread.sleep(MILLISECONDS.convert(sleepTime, sleepTimeUnit));
+    public void sleep(long sleepTimeInNano) throws InterruptedException {
+      // avoid throwing IllegalArgumentException
+      if (sleepTimeInNano > 0) {
+        Thread.sleep(NANOSECONDS.toMillis(sleepTimeInNano));
```

Comment:
I recommend testing the condition _after_ the conversion to milliseconds since e.g. `NANOSECONDS.toMillis(1) == 0`

Reply:
good point.